samza-core/src/main/scala/org/apache/samza/job/local/ThreadJobFactory.scala

```diff
@@ -54,18 +52,31 @@ class ThreadJobFactory extends StreamJobFactory with Logging {
       case _ => None
     }
 
+    val containerListener = new SamzaContainerListener {
+      override def onContainerFailed(t: Throwable): Unit = {
+        throw t
```

Comment:
Iiuc, ApplicationRunner is the user-facing api, even for thread jobs.

@xinyuiscool can you please clarify ?